org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -204,6 +212,12 @@ public boolean anyMatching() {
 		return getServers().stream().map(CompletableFuture::join).anyMatch(Objects::nonNull);
 	}
 
+	/**
+	 * Creates a new executor from this one, with the same settings and filter, so a further request can be run
+	 */
+	@Override
```

Comment:
I do not remember the details, but I remember that Effective Java (2nd Edition) recommends avoiding `clone` and use copy constructors instead, and I think it is a usually accepted pattern.
For example
`	public static LanguageServerDocumentExecutor forExecutor(final @NonNull LanguageServerDocumentExecutor executor) {`

and similar for the other executor types.

Could we follow that recommendation?

Reply:
If you want, but that's partly because `clone` and `Clonable` predate generics and are thus broken. I've not actually implemented the 'official' `Clonable` interface, just provided a method called `clone()` that is generic and thus properly covariantly typed.

Reply:
Alright, I see, I have overlooked that. I still think the copy constructor is preferable, as then there is still only pattern to get new executors (constructor) vs two pattern (close + constructor)